### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Deathlymad**
This moves the extended data and speficially the spacing matrices to the sample object and uses the sample to write files. As the property of the spacing is object bound. With the cache system it could cause lost data on copy or rename.

### CODE REVIEW

**Comment by Deathlymad** [comment]
es fehlt soweit ich sehe ein default fuer extended data um sample objekte zu erstellen.

**Review by muellerdo — CHANGES_REQUESTED** [review]
- More documentation on model (group) required.
- More code clean up
- Passing sample objects vs only required data to interface IOs
- load_details() function present in some interfaces, but never called + removed from abstract base class?

File: miscnn/model/model.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+
+class Model():
```
[the remark this answers is not in the document]
Comment (Deathlymad) [reply miscnn/model/model.py:3]: This was added to this pull request accidentally. I'll remove it.

File: miscnn/data_loading/data_io.py [hunk 2]
```diff
@@ -97,21 +97,19 @@ def sample_loader(self, index, load_seg=True, load_pred=False, backup=False):
         if load_pred:
             prediction = self.interface.load_prediction(index, self.output_path)
             sample.add_prediction(prediction)
-        # Add optional details to the sample object
-        sample.add_details(self.interface.load_details(index))
```
[the remark this answers is not in the document]
Comment (Deathlymad) [reply miscnn/data_loading/data_io.py:101]: Examples work.

**Comment by muellerdo** [comment]
Resampling

**Pull request merged**